### PR TITLE
Fix links in CONTRIBUTING.md.

### PR DESCRIPTION
**Pull request by houhoulis**
I am providing these changes to make it easier for a new contributor (like me) to find necessary information.

I have questions about 3 links in this file:

1. **"The Website" in the top section:** The Contributing guide doesn't exist, and the repo (https://github.com/exercism/website) is not accepting contributions. I linked to the README in case the website opens up in the future. Was that the right choice?
2. **"The Exercises API" in the top section:** I found a link that resolves, but is the API not a thing? Should I remove this line?
3. **"Test Data Versioning":** There is text in `CONTRIBUTING.md` that is not on https://github.com/exercism/problem-specifications/blob/main/CONTRIBUTING.md about test data versioning. It looks like that section is also in `README.md`, but not on https://github.com/exercism/problem-specifications/blob/main/README.md. Perhaps it's obsolete. I'm not sure how it can be in `main` but not display on GitHub dot com, so I don't know how or whether to fix that link. (It would be nice for these files and GitHub dot com to be consistent, for a newcomer.) The test data versioning link, which I didn't touch, is on line 192-193 in this PR. Should I change that link or delete those two sentences?

I am more than happy for any feedback or corrections, as I would like to use this guide to help me make a PR to add some tests to an exercise. Thanks.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:54]: I think you might have missed this bit at the top of this file:

> This guide covers several common scenarios pertaining to improving the language tracks themselves. There are other guides about contributing to other parts of the Exercism ecosystem. As such, the links are not specific to problem-specifications.

```suggestion
* open a [GitHub issue](https://github.com/exercism/exercism/issues).
```

I do agree that this is weird, and we'll likely move much of this content to our docs at some point.

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:67]:
```suggestion
[Code of Conduct](https://exercism.org/docs/using/legal/code-of-conduct).
```

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:20]: We currently don't plan on allow contributions to the website, so maybe remove this link for now?

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:21]: I think we should also remove this link, as the information seems to be quite outdated. Agree @iHiD?

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:126]:
```suggestion
in [this repository](https://github.com/exercism/problem-specifications).
```

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:237]: This link also points to a document with outdated information. It's probably a little bit better than nothing, but we should really be adding this information to our docs website :)

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:324]: We no longer have versioning for exercise, so this section can be removed (I'd do that in a separate PR though).

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:353]:
```suggestion
See the [Practice Exercises](https://exercism.org/docs/building/tracks/practice-exercises) page for more information on how to structure an exercise.
```

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:384]:
```suggestion
For more details on how to have the [configlet tool](https://exercism.org/docs/building/configlet) check your track via its linting functionality, see [this guide](https://exercism.org/docs/building/configlet/lint).
```

**Review by ErikSchierboom — COMMENTED** [review]
Thanks for working on this! In short: we're behind on porting our documentation. I expect most of the files that are linked to be integrated in the [docs part of the website](https://exercism.org/docs/). For now, we should link to new documentation when it exists, remove links that are not useful at the moment, and continue linking to old documents when it makes sense.

I've left some comments.

**Comment by houhoulis** [comment]
@ErikSchierboom I'm happy to make all these changes. Should I wait for @iHiD to answer your question? (https://github.com/exercism/problem-specifications/pull/1841#discussion_r712810675)

I'll have to rebase as well, since there seem to be conflicts.

File: CONTRIBUTING.md
Comment (iHiD) [reply CONTRIBUTING.md:21]: Agreed

**Comment by ErikSchierboom** [comment]
@houhoulis There was an answer by @iHiD, so I think you can update now.

**Comment by ErikSchierboom** [comment]
@houhoulis Are you still interested in working on this PR?

**Comment by houhoulis** [comment]
Sorry @ErikSchierboom , and thanks for the opportunity/reminder to make the necessary changes. I will do so this weekend.

**Comment by ErikSchierboom** [comment]
Great!

**Comment by houhoulis** [comment]
@ErikSchierboom I believe I addressed all of your feedback -- as of September. Sorry for the delay. Thanks!

**Comment by wolf99** [comment]
Ah, This PR may need to be adjusted with respect to #1895 , or vice-versa, depending on peoples preference ?

**Comment by SleeplessByte** [comment]
@wolf99 would you be okay to update that after we merge this, given that this is content fixes and that is formatting?

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:121]: I think maybe be specific here?

```suggestion
in [the exercism/problem-specifications repository](https://github.com/exercism/problem-specifications).
```

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:327]:
```suggestion
### Track-level Linting with configlet
```

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:329]:
```suggestion
For more details on how to have the [configlet tool](https://exercism.org/docs/building/configlet) verify your track via its linting functionality, see [this guide](https://exercism.org/docs/building/configlet/lint).
```

File: CONTRIBUTING.md
Comment (ErikSchierboom) [inline CONTRIBUTING.md:333]:
```suggestion
For help with Git, see [Git Basics](https://github.com/exercism/legacy-docs/blob/main/contributing/git-basics.md)
```

**Review by ErikSchierboom — APPROVED** [review]
Just a couple of minor tweaks. I'm approving in advance.

**Comment by houhoulis** [comment]
@ErikSchierboom Done. Thanks for the feedback.

**Review by ErikSchierboom — APPROVED** [review]

**Review by kotp — APPROVED** [review]
Looks good.  Thanks for the effort!

**Review by SleeplessByte — APPROVED** [review]

**Pull request merged**
